src/list.rs

```diff
@@ -0,0 +1,202 @@
+//! A [`List`] object represents a requested sequence of other objects (Cards,
+//! Sets, etc). List objects may be paginated, and also include information
+//! about issues raised when generating the list.
+//!
+//! This module also defines [`ListIter`], which can iterate over the contents
+//! of a `List`. If the list is paginated, the `ListIter` will request each page
+//! lazily.
+
+use std::vec;
+
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+
+use crate::uri::Uri;
+
+/// A list object.
+///
+/// For documentation on its fields refer to the [list object](https://scryfall.com/docs/api/lists)
+/// on the official site.
+#[derive(Serialize, Deserialize, Clone, Eq, PartialEq, Hash, Debug)]
+#[allow(missing_docs)]
+pub struct List<T> {
+    pub data: Vec<T>,
+    pub has_more: bool,
+    pub next_page: Option<Uri<List<T>>>,
+    pub total_cards: Option<usize>,
+    pub warnings: Option<Vec<String>>,
+}
+
+impl<T: DeserializeOwned> List<T> {
+    /// Creates an iterator over all the pages of this list.
+    pub fn into_page_iter(self) -> PageIter<T> {
+        PageIter {
+            curr: Some(self),
+            page_num: 1,
+        }
+    }
+}
+
+impl<T: DeserializeOwned> IntoIterator for List<T> {
+    type IntoIter = ListIter<T>;
+    type Item = T;
+
+    fn into_iter(self) -> Self::IntoIter {
+        // `has_more` is assumed to be redundant.
+        debug_assert!(self.has_more == self.next_page.is_some());
```

Comment:
maybe you should have `has_more` as just a method that checks the option and remove it from the struct?

Reply:
I included it so the struct would have all the fields from the [List Object](https://scryfall.com/docs/api/lists) page in the docs.